services/dumper.js

```diff
@@ -200,23 +200,21 @@ function Dumper(config) {
     });
 
     const referencesDefinition = references.map((reference) => {
-      const isBelongsToMany = reference.association === 'belongsToMany';
+      const computedReference = {
+        ...reference,
+        isBelongsToMany: reference.association === 'belongsToMany',
+      };
+
+      if (reference.junctionTable) {
+        computedReference.junctionTable = _.camelCase(reference.junctionTable);
+      }
 
       if (reference.targetKey) {
-        const expectedConventionalTargetKeyName = underscored
-          ? _.snakeCase(reference.targetKey) : _.camelCase(reference.targetKey);
-        const targetKeyColumnUnconventional = reference.targetKey
-          !== expectedConventionalTargetKeyName;
-        return {
-          ...reference,
-          isBelongsToMany,
-          targetKeyColumnUnconventional,
-        };
+        computedReference.targetKeyColumnUnconventional = reference.targetKey
+          !== (underscored ? _.snakeCase(reference.targetKey) : _.camelCase(reference.targetKey));
```

Comment:
The previous code was already hard to read.
Please don't "compact" it for the sake of optimisation, "beauty",...

Reply:
That was mostly for performance reasons (unnecessary variable creation), but I'll rollback.

Reply:
I do agree with Arnaud, variable creation is an acceptable performance drop to gain in code readability

Reply:
Fixed